openquake/hazardlib/gsim/kotha_2019.py

```diff
@@ -276,60 +279,133 @@ def get_stddevs(self, C, stddev_shape, stddev_types, sites):
+    8.000   1.063213165919580   3.450720620500530   0.427534350267617    1.454576290963340   -1.323042041332830   0.291715072522946   -0.064115203294262   0.151582553207261   0.507211779986322   0.422147964378804   0.234785059413908   0.386735725945679   -0.166955752594165   -0.176106622570061    0.195250506134964   0.315321534521396   -0.036026205230018   -0.145317983302120   -0.002646907312481   0.410746028675711
     """)
 
-    CONSTANTS = {"Mref": 4.5, "Rref": 30., "Mh": 6.2}
+    CONSTANTS = {"Mref": 4.5, "Rref": 30., "Mh": 6.2,
+                 "h_D10": 4.0, "h_10D20": 8.0, "h_D20": 12.0}
+
+
+class KothaEtAl2019Site(KothaEtAl2019):
+    """
+    Preliminary adaptation of the Kotha et al. (2019) GMPE using
+    a polynomial site amplification function dependent on Vs30 (m/s)
+    """
+    #: Required site parameter is not set
+    REQUIRES_SITES_PARAMETERS = set(("vs30",))
+
+    def get_site_amplification(self, C, sites, imt):
+        """
+        Defines a second order polynomial site amplification model
+        """
+        # Render with respect to 800 m/s reference Vs30
+        sref = np.log(sites.vs30 / 800.)
+        return C["g0_vs30"] + C["g1_vs30"] * sref + C["g2_vs30"] * (sref ** 2.)
+
+    def get_stddevs(self, C, stddev_shape, stddev_types, sites):
+        """
+        Returns the standard deviations
+        """
+        stddevs = []
+        tau = C["tau_event"]
+        phi = np.sqrt(C["phi0"] ** 2.0 + C["phis2s_vs30"] ** 2.)
+        for stddev_type in stddev_types:
+            assert stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+            if stddev_type == const.StdDev.TOTAL:
+                stddevs.append(np.sqrt(tau ** 2. + phi ** 2.) +
+                               np.zeros(stddev_shape))
+            elif stddev_type == const.StdDev.INTRA_EVENT:
+                stddevs.append(phi + np.zeros(stddev_shape))
+            elif stddev_type == const.StdDev.INTER_EVENT:
+                stddevs.append(tau + np.zeros(stddev_shape))
+        return stddevs
+
+
+class KothaEtAl2019Slope(KothaEtAl2019):
+    """
+    Preliminary adaptation of the Kotha et al. (2019) GMPE using
+    a polynomial site amplification function dependent on slope (m/m)
+    """
+    #: Required site parameter is not set
+    REQUIRES_SITES_PARAMETERS = set(("slope",))
+
+    def get_site_amplification(self, C, sites, imt):
+        """
+        Defines a second order polynomial site amplification model
+        """
+        # Render with respect to 0.1 m/m reference slope
+        sref = np.log(sites.slope / 0.1)
+        return C["g0_slope"] + C["g1_slope"] * sref +\
+            C["g2_slope"] * (sref ** 2.)
+
+    def get_stddevs(self, C, stddev_shape, stddev_types, sites):
```

Comment:
You are not using sites in this method.

Reply:
In other variants of the `.get_site_amplification(...)` method used in the subclasses one or both of sites and/or imt are needed. So I keep the call the the function general for all cases in the main class even if not all the terms are used in the subclasses.